### ISSUE TITLE
Export Module Versions

### ISSUE THREAD

**Issue by rainman110** [dlr-gtlab/gtlab-core#613]
In GitLab by @bjoernTWK on Nov 7, 2023, 15:12

## What is the purpose of the change?
- In projects different Versions of GTlab were used. For an easy documentation of the used GTlab-Modules an export button would be helpful, which exports a list of all loaded GTlab-Modules with their versions.

## Which tasks have to be performed?
- [ ] Add an export button

![grafik](https://my-gitlab-bucket.s3.amazonaws.com/727352741/2525a1daf2e0354173bec8658781011ceadbfc5927d8f944b5e0c5be10c3d775/grafik.png)

**Comment by jensschmeink** [comment]
@bjoernTWK do you think about a kind of exporter? Then the user could select the folder to save the file. It's a matter of taste, but I think a xml file is at least as readable as a csv file

**Comment by bjoernTWK** [comment]
> @bjoernTWK do you think about a kind of exporter? Then the user could select the folder to save the file. It's a matter of taste, but I think a xml file is at least as readable as a csv file

In my opinion, a csv file significantly simplifies the storage of versions. This means that the version does not have to be copied manually from the xml, especially for pure users (maybe an additional central function in the core would be helpful, where the versions of the loaded modules are returned as a map object, for example -> in this context, also the access via python would be interesting for the realization individuell solutions).

**Comment by jensschmeink** [comment]
I like the idea of a clear interface in the code to ask for the version setup of the modules of the application. I'm also fine with the idea of a csv file.

**Comment by rainman110** [comment]
Isn't there the footprint command of the console application. It should contain everything you need

**Comment by bjoernTWK** [comment]
> Isn't there the footprint command of the console application. It should contain everything you need

I'm not sure what is intention of the footprint function -> In GTlab-GUI there is only a hash shown, I would like to know which Module Versions are currently used
![image](https://github.com/dlr-gtlab/gtlab-core/assets/158561198/5381c81f-a3e6-43a0-8f5b-24b8e653b409)

**Comment by rainman110** [comment]
No, I mean the console application. It should show all modules and each module's version.

**Comment by jensschmeink** [comment]
@bjoernTWK Can I summerize your request as following: 
Please offer another data type (csv) for the export of GTlab's footprint?  
Currently the XML footprint export is offered in the "About GTlab" Window

![grafik](https://github.com/dlr-gtlab/gtlab-core/assets/148551711/bac27d32-104a-40b2-a4c6-d33e600b8c7b)

**Comment by bjoernTWK** [comment]
Additional I would like to have the possibility to get the information in a map-format -> so the user can get the information in all modules and within the python task

**Comment by jensschmeink** [comment]
> Additional I would like to have the possibility to get the information in a map-format -> so the user can get the information in all modules and within the python task

So the user could ask for it via the python interface and developers could access this in the code?

**Comment by rainman110** [comment]
I don't want to interrupt here. We have a feature to export all module information via the export footprint into xml format.

You can parse these and get out what you need. I don't like duplicating the same functionality, just to have another way of accessing the data.

We could discuss though, whether the footprint should be XML or in another more friendly way.

**Comment by jensschmeink** [comment]
@bjoernTWK What is your specific use case? Then we might understand how we could solve your problem

**Comment by bjoernTWK** [comment]
I like to create an overview of the used tool versions for engine results (in my case I provide the results within csv-files not as project files). Therefore, I like to add the DLRp2-version in my python export script and add to the result files an tool overview file

**Comment by jensschmeink** [comment]
As the foot print of module versions is offered in python interface the feature is not needed in  core anymore
